### PR TITLE
fix: display missing custom metrics and dimensions

### PR DESCRIPTION
**Pull request by almeidabbm**
Closes: #10066 

### Description:

https://github.com/lightdash/lightdash/assets/22939015/9206e91f-ac3f-4bba-9ffa-b3f323e466db

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/Explorer/ExploreTree/index.tsx [hunk 1]
```diff
@@ -71,12 +72,30 @@ const ExploreTree: FC<ExploreTreeProps> = ({
...
-        const allTables = Object.keys(explore.tables);
-        return additionalMetrics.filter(
-            (metric) => !allTables.includes(metric.table),
-        );
+        return additionalMetrics.filter((metric) => {
+            const table = explore.tables[metric.table];
+            return (
+                !table ||
+                (metric.baseDimensionName &&
+                    !table.dimensions[metric.baseDimensionName])
```
Comment (rephus) [inline packages/frontend/src/components/Explorer/ExploreTree/index.tsx:80]: why did you change this ? I thought the problem was only on custom dimensions

File: packages/frontend/src/components/Explorer/ExploreTree/index.tsx [hunk 2]
```diff
@@ -71,12 +72,30 @@ const ExploreTree: FC<ExploreTreeProps> = ({
...
+        });
     }, [explore, additionalMetrics]);
 
+    const missingCustomDimensions = useMemo(() => {
+        return customDimensions?.filter((customDimension) => {
+            const table = explore.tables[customDimension.table];
+
+            return (
+                !table ||
+                (isCustomBinDimension(customDimension) &&
```
Comment (rephus) [inline packages/frontend/src/components/Explorer/ExploreTree/index.tsx:91]: do we also need to check if `isCustomSqlDimension` ?

File: packages/frontend/src/components/Explorer/ExploreTree/index.tsx (hunk 1 above)
Comment (almeidabbm) [reply packages/frontend/src/components/Explorer/ExploreTree/index.tsx:80]: yeah, ended up noticing as well that if the base dimension for a custom metric was removed the custom metric wouldn't show as missing

https://github.com/lightdash/lightdash/assets/22939015/b396e478-3739-4077-aeb0-c6d6fe3b3436

File: packages/frontend/src/components/Explorer/ExploreTree/index.tsx (hunk 2 above)
Comment (almeidabbm) [reply packages/frontend/src/components/Explorer/ExploreTree/index.tsx:91]: good point, I had a look at that and `customSqlDimension` but since it doesn't have a base dimension and it's only sql I couldn't do it like this, probably can only do it after compiling?

Comment (almeidabbm) [reply packages/frontend/src/components/Explorer/ExploreTree/index.tsx]: I believe I was able to do this in the latest [commit](https://github.com/lightdash/lightdash/pull/10070/commits/f8dd428263461d83a9f87cf9f99776c92496bbc1): 


https://github.com/lightdash/lightdash/assets/22939015/dfc2232b-fb50-4289-baca-4f5ebed702a7

**Review by rephus — APPROVED** [review]
